src/components/PublishCarForm.component.tsx

```diff
@@ -0,0 +1,306 @@
+import React, { useEffect } from "react";
+import { SubmitHandler, useForm } from "react-hook-form";
+import { toast } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+import styled from "styled-components";
+
+import {
+  useCreateCarMutation,
+  useFormDataQuery,
+  useModelsLazyQuery,
+} from "../graphql/generated/graphql";
+import { modelsbyBrandIdVariables } from "../graphql/variables";
+import { GlobalStyle, defaultTheme } from "../theme";
+import { ConditionFieldset } from "./ConditionFieldset";
+import { FormSelectInput } from "./FormSelectInput.component";
+import {
+  ButtonOnHoverOppacity,
+  CSSFlexCol,
+  FlexRow,
+  Form,
+  H2,
+  H3,
+  InputStyle,
+  P,
+  ValidationText,
+} from "./styled";
+import Loader from "./styled/Loader.component";
+
+interface FormValues {
+  brand_id: number;
+  model_id?: number;
+  year?: number;
+  state_id?: number;
+  city_id?: number;
+  vin?: string;
+  color_id?: number;
+  odometer: number;
+  condition: string;
+  sale_date: string;
+  price: number;
+  title?: string;
+}
+const VALIDATION_MESSAGES = {
+  BRAND: "Select your car's Brand",
+  MODEL: "Select your car's Model",
+  YEAR: "Enter your car's fabrication year",
+  CITY: "Select your car's city of procedence",
+  VIN: "Enter your Vehicle's Identification Number",
+  COLOR: "Select your car's color",
+  CONDITION: "Pick your car's condition",
+  SALE_DATE: "Select when you want to sell your car",
+  PRICE: "Enter a price for your car",
+};
+export const PublishCarForm = () => {
+  const { data: formsData, error: dataError, loading } = useFormDataQuery();
+  const [
+    fetchModels,
+    { data: modelsData, error: modelsError, loading: modelsLoading },
+  ] = useModelsLazyQuery();
+
+  const [createCar, { data: mutationReturn, loading: mutationLoading }] =
+    useCreateCarMutation();
+  const {
+    register,
+    reset,
+    handleSubmit,
+    watch,
+    setValue,
+    getValues,
+    formState: { errors },
+  } = useForm<FormValues>();
+
+  const threeMonthsAhead = new Date();
+  threeMonthsAhead.setDate(threeMonthsAhead.getDate() + 90);
+
+  const brandId = watch("brand_id");
+  const odometer = watch("odometer");
+  const cityId = watch("city_id");
+
+  const onSubmit: SubmitHandler<any> = async (data) => {
+    const brand = formsData?.brands.find(
+      (brand) => brand.id == getValues("brand_id")
+    )?.name;
+    const model = modelsData?.models.find(
+      (model) => model.id == getValues("model_id")
+    )?.name;
+    data.title = `${brand} ${model} ${getValues("year")}`;
+    for (const key in data)
+      if (key.includes("id")) data[key] = parseInt(data[key]);
+
+    try {
+      await createCar({
+        variables: { object: data },
+        optimisticResponse: {
+          insert_cars_one: data,
+          __typename: "mutation_root",
```

Comment:
I don't think this is the typename for this mutation, where did you get this from?

Reply:
Yikes, you're right, in my case I'm handling the cache with `cache.update` that optimistic response didn't seem to affect how I handled the cache.